log4j-core/src/main/java/org/apache/logging/log4j/core/config/AbstractConfiguration.java

```diff
@@ -687,28 +687,37 @@ protected void doConfigure() {
             configurationStrSubstitutor.setVariableResolver(interpolator);
         }
 
+        Node scriptsNode = null;
+        for (final Node node : rootNode.getChildren()) {
+            if ("Scripts".equalsIgnoreCase(node.getName())) {
+                scriptsNode = node;
+                createConfiguration(scriptsNode, null);
+                if (scriptsNode.getObject() != null) {
+                    for (final AbstractScript script : scriptsNode.getObject(AbstractScript[].class)) {
```

Comment:
The `scriptsNode` variable is not really useful here. Otherwise it looks good to me.